openssl/src/x509/mod.rs

```diff
@@ -451,6 +451,15 @@ impl X509Ref {
         }
     }
 
+    /// Returns the X509 public key of the certificate request.
+    #[corresponds(X509_get_X509_PUBKEY)]
+    pub fn x509_public_key(&self) -> Result<X509PublicKey, ErrorStack> {
```

Comment:
This needs to return a `&X509PublicKeyRef` according to the OpenSSL documentation.